core/corehttp/gateway_handler.go

```diff
@@ -133,6 +133,7 @@ func (i *gatewayHandler) getOrHeadHandler(w http.ResponseWriter, r *http.Request
 	}
 
 	w.Header().Set("X-IPFS-Path", urlPath)
+	w.Header().Set("Access-Control-Allow-Origin", "*")
```

Comment:
We shouldn't set it always like this. this should be an option. Also, like the API (see `commands/http/handler.go`) we may have to set the incoming domain, as `*` doesn't work for certain things.